src/nvme/tree.c

```diff
@@ -1076,19 +1081,25 @@ nvme_ctrl_t __nvme_lookup_ctrl(nvme_subsystem_t s, const char *transport,
 
 	c = p ? nvme_subsystem_next_ctrl(s, p) : nvme_subsystem_first_ctrl(s);
 	for (; c != NULL; c = nvme_subsystem_next_ctrl(s, c)) {
-		if (strcmp(c->transport, transport))
```

Comment:
subsysname, subsysnqn, transport, traddr and host_traddr can never be null. Thus this change is not neccessary.

Reply:
I think that `host_traddr` can be null. Can't it? Or is it set to an empty string when not used?

Reply:
I mean the entries in the tree have these values always initialized to a valid string.